verifier/src/main/kotlin/tech/figure/classification/asset/verifier/client/VerifierClient.kt

```diff
@@ -87,18 +89,28 @@ class VerifierClient(private val config: VerifierClientConfig) {
         val currentHeight = config.eventStreamProvider.currentHeight()
         var latestBlock = startingBlockHeight?.takeIf { start -> start > 0 && currentHeight?.let { it >= start } != false }
 
+        suspend fun newBlock(height: Long) {
```

Comment:
Nit: Try to avoid declaring functions within functions in non-test code.  Please uplift this dealio into a private function outside the loop, unless there's some issue with doing that that I'm missing